### PR TITLE
Create a service level config framework and per user rate limits

### PR DESCRIPTION
**Pull request by bwillard**

### CODE REVIEW

File: portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/config/TransferConfig.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.datatransferproject.spi.transfer.config;
+
+public interface TransferConfig {
```
Comment (holachuy) [inline portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/config/TransferConfig.java]: java doc

File: portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java [hunk 2]
```diff
@@ -197,4 +206,22 @@ Monitor getMonitor() {
...
+
+  private TransferServiceConfig getTransferServiceConfig(TransferExtension ext) {
+    String configFileName = "config/" + ext.getServiceId().toLowerCase() + ".yaml";
+    InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream(configFileName);
+    getMonitor().info(
+        () -> "Service %s has a config file: %s",
+        ext.getServiceId(),
+        (inputStream != null));
+    if (inputStream == null) {
+      return TransferServiceConfig.getDefaultInstance();
```
Comment (holachuy) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java]: Let's log: using default transfer config for serviceId ...

Comment (bwillard) [reply portability-transfer/src/main/java/org/datatransferproject/transfer/WorkerModule.java]: This is logged about the if statement.

File: portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/config/TransferConfig.java (hunk 1 above)
Comment (bwillard) [reply portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/config/TransferConfig.java]: I think we can just delete this file.

**Comment by bwillard** [comment]
Thanks, added https://github.com/google/data-transfer-project/issues/694

**Pull request merged**